Assets/Scripts/Camera/CameraController.cs

```diff
@@ -51,46 +53,38 @@ void HandleMouseInput()
             newZoom += Input.mouseScrollDelta.y * zoomAmount;
         }
 
-        if(Input.GetMouseButtonDown(0))
+        if(Input.GetButtonDown("Mouse Right"))
         {
-            Plane plane = new Plane(Vector3.up, Vector3.zero);
-
-            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
-        
-            float entry;
+            var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
 
-            if(plane.Raycast(ray, out entry))
+            if(plane.Raycast(ray, out var entry))
             {
                 dragStartPosition = ray.GetPoint(entry);
             }
         }
 
-        if(Input.GetMouseButton(0))
+        if(Input.GetButton("Mouse Right"))
         {
-            Plane plane = new Plane(Vector3.up, Vector3.zero);
-
-            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
-        
-            float entry;
+            var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
 
-            if(plane.Raycast(ray, out entry))
+            if(plane.Raycast(ray, out var entry))
             {
                 dragCurrentPosition = ray.GetPoint(entry);
 
                 newPosition = transform.position + dragStartPosition - dragCurrentPosition;
             }
         }
 
-        if(Input.GetMouseButtonDown(2))
+        if(Input.GetButtonDown("Mouse Middle"))
```

Comment:
It will be really hard to control on a Mac mouse. You don't have to fix it now, but you should create task for further fixing.